lib/pure/collections/critbits.nim

```diff
@@ -401,29 +423,14 @@ proc allprefixedAux[T](c: CritBitTree[T], key: string;
         if i >= p.key.len or p.key[i] != key[i]: return
     result = top
 
-iterator itemsWithPrefix*[T](c: CritBitTree[T], prefix: string;
-                             longestMatch = false): string =
-  ## Yields all keys starting with `prefix`. If `longestMatch` is true,
-  ## the longest match is returned, it doesn't have to be a complete match then.
-  runnableExamples:
-    var c: CritBitTree[int]
-    c["key1"] = 42
-    c["key2"] = 43
-    var keys: seq[string]
-    for key in c.itemsWithPrefix("key"):
-      keys.add(key)
-    doAssert keys == @["key1", "key2"]
-
-  let top = allprefixedAux(c, prefix, longestMatch)
-  for x in leaves(top): yield x.key
-
 iterator keysWithPrefix*[T](c: CritBitTree[T], prefix: string;
                             longestMatch = false): string =
   ## Yields all keys starting with `prefix`.
+  ##
+  ## If `longestMatch` is true and no key starts with `prefix`, only
+  ## the longest match is yielded, it doesn't have to be a complete match then.
```

Comment:
can you add an example for longestMatch=true?

Reply:
Hmm, now I'm actually confused how `longestMatch` is supposed to work:
```nim
let c = {"key1": 1, "key2": 2}.toCritBitTree
echo toSeq(c.keysWithPrefix("lol", longestMatch = true))  # => @["key1", "key2"]
echo toSeq(c.keysWithPrefix("keys", longestMatch = true)) # => @["key2"]
```
Apparently it doesn't always yield every key with a longest match, but it doesn't always only yield a single element either...

Reply:
looks super buggy, can you please open an issue to track this? and in your PR, just skip the runnableExamples for now until we figure out what to do here

Reply:
Done: #16575.